src/lib/arch/aarch64/Architecture.cc

```diff
@@ -188,6 +188,20 @@ uint8_t Architecture::predecode(const uint8_t* ptr, uint16_t bytesAvailable,
     newInsn.setExecutionInfo(getExecutionInfo(newInsn));
     // Cache the instruction
     iter = decodeCache_.insert({insn, newInsn}).first;
+  } else {
```

Comment:
For non Predicate / SVE instructions the overhead is 3 if statements and a function call which _should_ be minor. But yes, a performance regression test for this would be good. 

Not sure on what an alternative solution could be though if the performance impact is significant...

Reply:
I'll leave this comment unresolved so we remember to look into this